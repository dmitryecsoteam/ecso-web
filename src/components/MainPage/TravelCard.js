import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default class TravelCard extends PureComponent {

    render() {
        const {
            _id,
            name,
            country,
            nameEn,
            countryEn,
            priceAirplane,
            carDistance,
            parameters,
            originName,
            date
        } = this.props;

        const parametersList = [];
        for (let param in parameters) {
            parametersList.push(<span className="travel-card__text--medium travel-card__text--padding-right" key={parametersList.length}>{`${param}: ${parameters[param]}`}</span>);
        }

        let priceInfo;
        if (!priceAirplane && carDistance <= 200) {
            priceInfo = (<div className="travel-card__price">
                <span className="travel-card__text--small">by car</span>
                <span className="travel-card__text--big">{`approx. ${carDistance} km`}</span>
            </div>);
        } else {
            priceInfo = (<div className="travel-card__price">
                <span className="travel-card__text--small">airplane</span>
                <span className="travel-card__text--big">{priceAirplane ? `from ${priceAirplane} $` : `find prices`}</span>
            </div>);
        }

        const originAndDate = (
            <p className="travel-card__text--big">From {originName} &emsp; {date}</p>
        );

        return (
            <Link
                to={`/travel/${_id}`}
                className="travel-card"
            >

                <div
                    className="travel-card__image"
                    style={{ backgroundImage: `url(/images/${nameEn.replace(/ /g, '_')}-${countryEn}/banner.jpg)` }}
                >
                </div>

                <div className="travel-card__info">

                    <h2>
                        {`${name}, ${country}`}
                    </h2>

                    <div className="travel-card__parameters">
                        {originName ? originAndDate : parametersList}
                    </div>

                    {priceInfo}

                </div>
            </Link>)
    }
}

TravelCard.propTypes = {
    _id: PropTypes.string,
    name: PropTypes.string,
    country: PropTypes.string,
    nameEn: PropTypes.string,
    countryEn: PropTypes.string,
    priceAirplane: PropTypes.number,
    carDistance: PropTypes.number,
    parameters: PropTypes.object
}