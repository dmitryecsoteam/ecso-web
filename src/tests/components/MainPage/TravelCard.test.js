import React from 'react';
import { shallow } from 'enzyme';
import TravelCard from '../../../components/MainPage/TravelCard';

let wrapper;
beforeEach(() => {
    wrapper = shallow(<TravelCard
        _id="5c6c7e42e3b7e1edb974a39b"
        name="Osaka"
        country="Japan"
        nameEn="Osaka"
        countryEn="Japan"
        priceAirplane={123}
        carDistance={189}
        parameters={{}}
    />);
});

test('should render TravelCard with empty parameters', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render TravelCard with parameters', () => {
    wrapper.setProps({ parameters:{
        Beach: 4,
        Food: 5,
        Museum: 4
    } });
    expect(wrapper).toMatchSnapshot();
});

test('should render TravelCard with find airplane prices', () => {
    wrapper.setProps({ priceAirplane: null, carDistance: 225 });
    expect(wrapper).toMatchSnapshot();
});

test('should render TravelCard with car distance', () => {
    wrapper.setProps({ priceAirplane: null });
    expect(wrapper).toMatchSnapshot();
});