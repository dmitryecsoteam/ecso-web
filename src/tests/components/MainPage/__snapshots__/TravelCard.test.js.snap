// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TravelCard with car distance 1`] = `
<Link
  className="travel-card"
  replace={false}
  to="/travel/5c6c7e42e3b7e1edb974a39b"
>
  <div
    className="travel-card__image"
    style={
      Object {
        "backgroundImage": "url(/images/Osaka-Japan/banner.jpg)",
      }
    }
  />
  <div
    className="travel-card__info"
  >
    <h2>
      Osaka, Japan
    </h2>
    <div
      className="travel-card__parameters"
    />
    <div
      className="travel-card__price"
    >
      <span
        className="travel-card__text--small"
      >
        by car
      </span>
      <span
        className="travel-card__text--big"
      >
        approx. 189 km
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render TravelCard with empty parameters 1`] = `
<Link
  className="travel-card"
  replace={false}
  to="/travel/5c6c7e42e3b7e1edb974a39b"
>
  <div
    className="travel-card__image"
    style={
      Object {
        "backgroundImage": "url(/images/Osaka-Japan/banner.jpg)",
      }
    }
  />
  <div
    className="travel-card__info"
  >
    <h2>
      Osaka, Japan
    </h2>
    <div
      className="travel-card__parameters"
    />
    <div
      className="travel-card__price"
    >
      <span
        className="travel-card__text--small"
      >
        airplane
      </span>
      <span
        className="travel-card__text--big"
      >
        from 123 $
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render TravelCard with find airplane prices 1`] = `
<Link
  className="travel-card"
  replace={false}
  to="/travel/5c6c7e42e3b7e1edb974a39b"
>
  <div
    className="travel-card__image"
    style={
      Object {
        "backgroundImage": "url(/images/Osaka-Japan/banner.jpg)",
      }
    }
  />
  <div
    className="travel-card__info"
  >
    <h2>
      Osaka, Japan
    </h2>
    <div
      className="travel-card__parameters"
    />
    <div
      className="travel-card__price"
    >
      <span
        className="travel-card__text--small"
      >
        airplane
      </span>
      <span
        className="travel-card__text--big"
      >
        find prices
      </span>
    </div>
  </div>
</Link>
`;

exports[`should render TravelCard with parameters 1`] = `
<Link
  className="travel-card"
  replace={false}
  to="/travel/5c6c7e42e3b7e1edb974a39b"
>
  <div
    className="travel-card__image"
    style={
      Object {
        "backgroundImage": "url(/images/Osaka-Japan/banner.jpg)",
      }
    }
  />
  <div
    className="travel-card__info"
  >
    <h2>
      Osaka, Japan
    </h2>
    <div
      className="travel-card__parameters"
    >
      <span
        className="travel-card__text--medium travel-card__text--padding-right"
        key="0"
      >
        Beach: 4
      </span>
      <span
        className="travel-card__text--medium travel-card__text--padding-right"
        key="1"
      >
        Food: 5
      </span>
      <span
        className="travel-card__text--medium travel-card__text--padding-right"
        key="2"
      >
        Museum: 4
      </span>
    </div>
    <div
      className="travel-card__price"
    >
      <span
        className="travel-card__text--small"
      >
        airplane
      </span>
      <span
        className="travel-card__text--big"
      >
        from 123 $
      </span>
    </div>
  </div>
</Link>
`;
